### PR TITLE
feat(ui): Move Auth Providers list to react

### PR DESCRIPTION
**Pull request by billyvg**
* change Auth Provider configuration to be in "modal" template

Depends on https://github.com/getsentry/sentry/pull/7577

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/organization/auth/providerItem.jsx [hunk 1]
```diff
@@ -16,23 +19,43 @@ export default class ProviderItem extends React.PureComponent {
   };
 
   handleConfigure = e => {
-    this.props.onConfigure(this.props.providerKey, e);
+    if (typeof this.props.onConfigure === 'function') {
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/settings/organization/auth/providerItem.jsx]: why was this needed?

File: src/sentry/static/sentry/less/organization.less [hunk 2]
```diff
@@ -375,79 +375,6 @@
   }
 }
 
-/**
```
Comment (MaxBittker) [inline src/sentry/static/sentry/less/organization.less]: 👍

File: src/sentry/templates/sentry/organization-auth-settings.html [hunk 3]
```diff
@@ -1,39 +0,0 @@
-{% extends "sentry/bases/organization.html" %}
```
Comment (MaxBittker) [inline src/sentry/templates/sentry/organization-auth-settings.html]: 👍

File: src/sentry/static/sentry/app/views/settings/organization/auth/providerItem.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/organization/auth/providerItem.jsx]: 1) Being defensive
2) Change of API (at the time this component was added, we had planned to change the configuration form into react, but now we are leaving it in django)

File: src/sentry/web/frontend/organization_auth_settings.py [hunk 4]
```diff
@@ -202,20 +204,7 @@ def handle(self, request, organization):
...
-                continue
-
-            provider_list.append((k, v.name))
-
-        context = {
-            'provider_list': provider_list,
-        }
-
-        return self.respond('sentry/organization-auth-settings.html', context)
+        # Otherwise user is in bad state since frontend/react should handle this case
```
Comment (MaxBittker) [inline src/sentry/web/frontend/organization_auth_settings.py]: is this worth recording a sentry message for?

**Review by MaxBittker — APPROVED** [review]
this looks good to me, nice work 👍

**Comment by EvanPurkhiser** [comment]
Some issues (already mentioned some of this to @billyvg):

 * Styles on the list look a little wonky in terms of spacing
 * The 'Back to Organization' text on the modal is quite large for some reason.
 * After removing the auth method it redirects back to the old settings.
